### PR TITLE
backend: Grant update only after checking the version

### PR DESCRIPTION
**Pull request by pothos**
When the fast path was added to re-grant an update that was already
granted (6bcd6fdbfd2d7ad29b17ab830219ed1aad34be1a) this skipped the
version check, resulting in granting an update to clients that already
run this version. The DB state of granting an update should not be
more important than the version of the instance because the DB can either
lack entries (due to the client not sending them or the server dropping
them) or it can also be tricked by a malicious third party pretending
to be the client but with a lower app version.
Only re-grant an update after performing all checks that are normally
done as well, i.e., if the client's version needs an update and if the
package wasn't forbidden for this channel in the mean time. Set the
instance status to InstanceStatusComplete to keep the right entry in the
log and also don't allow the instance to bypass the rollout policy for a
future update.

### CODE REVIEW

**Comment by krnowak** [comment]
Would you mind adding a test that triggers the problem you fixed? Thanks.

File: pkg/api/updates.go [hunk 1]
```diff
@@ -79,22 +79,28 @@ func (api *API) GetUpdatePackage(instanceID, instanceIP, instanceVersion, appID,
...
 	}
 
-	if updateAlreadyGranted {
-		return group.Channel.Package, nil
-	}
-
 	for _, blacklistedChannelID := range group.Channel.Package.ChannelsBlacklist {
 		if blacklistedChannelID == group.Channel.ID {
+			if updateAlreadyGranted {
+				_ = api.updateInstanceStatus(instance.ID, appID, InstanceStatusComplete)
```
[the remark this answers is not in the document]
Comment (pothos) [reply pkg/api/updates.go:86]: Ok, added the comment

**Comment by pothos** [comment]
Did a `fmt --width=72` for the commit message

**Comment by joaquimrocha** [comment]
> Did a `fmt --width=72` for the commit message

Okay, Github made me think it was not correctly formatted but checking the actual branch shows it's fine. Sorry for the noise.

**Comment by pothos** [comment]
I should have said `Now I reformated with …`, you saw correctly that it wasn't aligned on 72 characters ;)

**Pull request merged**